libs/adapters/src/aws/index.ts

```diff
@@ -0,0 +1,82 @@
+import {
+  CompleteMultipartUploadCommandOutput,
+  PutObjectCommand,
+  S3,
+} from '@aws-sdk/client-s3';
+import { Upload } from '@aws-sdk/lib-storage';
+import { getSignedUrl } from '@aws-sdk/s3-request-presigner';
+import { BlobBucket, type BlobStorage } from '@hicommonwealth/core';
+import { config } from '../config';
+
+const s3Buckets: Record<BlobBucket, string> =
+  config.APP_ENV === 'local'
+    ? {
+        assets: 'local.assets',
+        archives: 'local.outbox-event-stream-archive',
+        sitemap: 'local.sitemap',
+      }
+    : {
+        assets: 'assets.commonwealth.im',
+        archives: 'outbox-event-stream-archive',
+        sitemap: 'sitemap.commonwealth.im',
+      };
+
+function formatS3Url(data: CompleteMultipartUploadCommandOutput) {
+  return config.APP_ENV === 'local'
+    ? `https://s3.amazonaws.com/${data.Bucket}/${data.Key}`
+    : `https://${data.Bucket}/${data.Key}`;
```

Comment:
For the non-local value, is it missing the base url?

Reply:
Also, is there a reason we're checking which environment it's running? Ideally most code shouldn't care.

Reply:
> Also, is there a reason we're checking which environment it's running? Ideally most code shouldn't care.

We created segregated buckets for local dev, Heroku staging, and production environments so that we're not putting everything into a single bucket and we can periodically clear local/staging buckets instead of indefinitely paying for that storage.

Reply:
Having segregated buckets is fine– my concern was about the code checking if it's running locally or not. That's generally an antipattern because it leads to potentially different behavior on local vs prod.

For example, in this case, the local url and prod url make different assumptions about the value of `data.Bucket`. Locally, it's the bucket name, but on prod, it'll be the base URL + bucket name? That looks wrong.

It would make more sense for there to be a single string: `https://${baseUrl}/${data.Bucket}/${data.Key}`– not sure where baseUrl would come from, but that pattern makes more sense because it's environment agnostic.

Reply:
Product bucket names are behind a CDN i.e. assets.commonwealth.im whereas local dev buckets are not.

Reply:
Understood, that's why I mentioned having a `baseUrl` variable. But that doesn't address my concern about `data.Bucket`.

Reply:
@rbennettcw we have CDN mappings using those data.Bucket names, the bucket is the base url.

We can move bucket names to config, but that's probably another PR... this has been always hardcoded

Reply:
@Rotorsoft I think `https://${data.Bucket}/${data.Key}` will break in production for the `outbox-event-stream-archive` bucket since that bucket is _not_ behind a CDN so it's not resolveable by itself.

Reply:
Adding archive bucket mapping rule here [6d00c96](https://github.com/hicommonwealth/commonwealth/pull/8666/commits/6d00c9626f77c0e105c5f055126f76e44383a46a)